File: task-03/Ruby/Sub-Task3.rb
# Function to generate and print the diamond pattern
def print_diamond(n)
  # Top half including the middle line
  (0...n).each do |i|
    # Print leading spaces
    print ' ' * (n - i - 1)
    # Print asterisks
    print '*' * (2 * i + 1)
    # Move to the next line
    puts
  end

  # Bottom half excluding the middle line
  (n - 2).downto(0) do |i|
    # Print leading spaces
    print ' ' * (n - i - 1)
    # Print asterisks
    print '*' * (2 * i + 1)
    # Move to the next line
    puts
  end
end

# Prompt user for input
print 'Enter the number of rows for the diamond pattern: '
n = gets.to_i

# Validate the input
if n > 0
  # Generate and print the diamond pattern
  print_diamond(n)
else
  puts 'Please enter a positive integer.'
end

File: task-03/Javascript/Sub-Task3.js
const readline = require('readline');

// Create an interface for input and output
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

// Function to generate and print the diamond pattern
function printDiamond(n) {
    // Top half including the middle line
    for (let i = 0; i < n; i++) {
        let line = '';
        
        // Print leading spaces
        for (let j = 0; j < n - i - 1; j++) {
            line += ' ';
        }
        
        // Print asterisks
        for (let j = 0; j < 2 * i + 1; j++) {
            line += '*';
        }
        
        console.log(line);
    }

    // Bottom half excluding the middle line
    for (let i = n - 2; i >= 0; i--) {
        let line = '';
        
        // Print leading spaces
        for (let j = 0; j < n - i - 1; j++) {
            line += ' ';
        }
        
        // Print asterisks
        for (let j = 0; j < 2 * i + 1; j++) {
            line += '*';
        }
        
        console.log(line);
    }
}

// Prompt the user for input
rl.question('Enter the number of rows for the diamond pattern: ', (answer) => {
    const n = parseInt(answer, 10);

    if (isNaN(n) || n <= 0) {
        console.log('Please enter a positive integer.');
    } else {
        printDiamond(n);
    }

    // Close the readline interface
    rl.close();
});
